### PR TITLE
Consistent plotstyle

### PR DESCRIPTION
**Pull request by chroetz**
Extended number of fixed colors for unknown entities in `plotstyle()` from 9 to 19.
Before, if you show two plots, e.g., plots for *mainReg* and *other regions* in compareScenarios, with different numbers of unknown entities (scenarios + historic data) where one number is larger than 9, the colors differed between the plots. This could lead to two plots on the same page showing different colors for your scenarios. 
The threshold number for this effect is now raised to 19. Furthermore, the user is warned when more then 19 colors for unknown entities are chosen. Raising the number further does not seem necessary, as more than 19 entities in a plot can hardly be distinguished at all, no matter the choice of colors. 
The 19 colors are shown in the documentation of `plotstyle()`.

The relevant changes are in the block that starts [here](https://github.com/pik-piam/mip/pull/31/files/2a1e3741111bf6fadc98df27009088e4945f9af0#diff-07ad1ca921f14d2abe89235e4af942b539f23492240b822243fd0069b6d4a240L107) and are committed [here](https://github.com/pik-piam/mip/pull/31/commits/dea7135b8acd5eee04cb7bfe2aa5791d2d96479a) and [here](https://github.com/pik-piam/mip/pull/31/commits/cc1d46c131c5c2f518d9de93b0e36e5a4a20171e).

### CODE REVIEW

File: R/plotstyle.R [hunk 1]
```diff
@@ -1,174 +1,194 @@
...
+#' entities <- c("AFR", "AAA", "AFR", "UNKNOWN_ELEMENT2")
 #' plotstyle(entities)
-#' unknown=data.frame(row.names=c("AAA","UNKNOWN_ELEMENT2"),
-#'                    color=c("#123456","#345678"),
-#'                    legend=c("l_AAA","l_UNKNOWN_ELEMENT2"))
-#' plotstyle(entities,unknown=unknown)
-#' plotstyle(entities,out="legend")
-#' plotstyle(entities,out="all")
-#' 
+#' unknown <- data.frame(row.names = c("AAA", "UNKNOWN_ELEMENT2"),
```
Comment (orichters) [inline R/plotstyle.R:32]: thanks for all this 'beauty' work – but it sometimes makes it a bit difficult to see where you actually changed something in the code ;)

File: R/plotstyle.R [hunk 2]
```diff
@@ -1,174 +1,194 @@
...
+    row.names = 1)
+
+  if (is.null(out)) {
     out <- "color"
-  } else if(!(out %in% c(names(luplot$plotstyle),"all"))) {
-    stop('Unknown style type "',out,'"!')
+  } else if (!(out %in% c(names(luplot$plotstyle), "all"))) {
+    stop('Unknown style type "', out, '"!')
   }
-
```
Comment (orichters) [inline R/plotstyle.R:50]: Is there a reason you don't allow the gray levels anymore?

File: NAMESPACE [hunk 3]
```diff
@@ -21,6 +21,8 @@ export(showRegiLinePlots)
 export(sideBySidePlots)
 export(theme_mip)
 export(validationpdf)
+import(magclass)
```
[the remark this answers is not in the document]
Comment (chroetz) [reply NAMESPACE:24]: Hm, this was done automatically by `lucode2::buildLibrary()`, I think. So, I assume it is ok.

File: R/plotstyle.R (hunk 1 above)
Comment (chroetz) [reply R/plotstyle.R:32]: I tried to follow the coding guideline to keep to the style guide and potentially fix styling issues as indicated by `lucode2::lint()`, e.g., by using `lucode2::autoFormat()`. But I see your point... 
Ideally, style is only improved and new code is committed with good code style. Then there will only be a finite number of such confusing style fixing commits ;-)

File: R/plotstyle.R (hunk 2 above)
Comment (chroetz) [reply R/plotstyle.R:50]: The function `.makegray()` is never used in the code. So I removed it.

File: R/plotstyle.R [hunk 4]
```diff
@@ -1,174 +1,194 @@
...
-      # if you need less than 10 colors brew and ramp for the exact number of colors min(9,nna) -> nna (ramping has no effect -> take the original brewer colors)
-      # only if you need more than 9 colors start ramping between maximal 9 brewer colors effectively
-      # replace NA in color with freshly brewed colors from "Set1"
-      tmpcols <- brewer.pal(9,"Set1")
-      tmpcols[6] <- "#00CED1" #replace yellow with turquoise as it was to light for many applications
-      if(nna<9) tmpcols <- tmpcols[1:nna]
-      set1 <- colorRampPalette(tmpcols)
+      # The following list contains 19 easily distinguishable colors. If you
+      # need n < 20 colors, choose the first n colors from that list. If you
+      # need more colors, colorRampPalette() is used to create interpolated
```
Comment (orichters) [inline R/plotstyle.R:118]: still, if you have 20 or more colors, the first 19 are not stable, but the question is whether this bothers someone. With 20 lines, you probably have other problems :)

An alternative could be to define `colorPalette` as:
``` R
colorPalette <- function(n) {
  return(c(tmpcols[1:min(n, length(tmpcols))], heat.colors(max(0, n-length(tmpcols)))))
}
```
This way, the first 19 colors are never changed and simply further colors are added, and only those are not stable.

**Review by orichters — APPROVED** [review]

File: R/plotstyle.R (hunk 1 above)
Comment (chroetz) [reply R/plotstyle.R:32]: Good point! I added links in the initial description.

File: R/plotstyle.R (hunk 4 above)
Comment (chroetz) [reply R/plotstyle.R]: Good idea! Inspired by you comment, I change it so that it is always consistent. But maybe this was overkill... 😁

Comment (orichters) [reply R/plotstyle.R]: I can confirm that it is certainly overkill. A pity that you made it deterministic :)

**Pull request merged**